core/retry_failed.py

```diff
@@ -0,0 +1,63 @@
+from dbt.config import RuntimeConfig
+
+import os
+import json
+import subprocess
+
+RUN_RESULTS_FILE = "run_results.json"
+
+
+class RetryFailedTask:
+    def __init__(self, args):
+        self.args = args
+        self.config = RuntimeConfig.from_args(args)
+        self.target_path = self.config.target_path
+        self.run_results = self._get_run_results()
+
+    def _get_run_results(self):
+        """
+        This subcommand parses the results.json file
+        """
+        run_results_path = os.path.join(self.target_path, RUN_RESULTS_FILE)
+        try:
+            with open(run_results_path) as f:
+                run_results = json.load(f)
+            return run_results
+        except IOError:
+            raise Exception("Could not find {} file.".format(RUN_RESULTS_FILE))
+
+    def get_models_to_retry(self):
+        """ Return a list of errored and skipped models
+        """
+        models = []
+        for result in self.run_results.get("results"):
+            if result["status"] == "ERROR" or result["skip"]:
+                models.append(result["node"]["name"])
+        return models
+
+    def get_run_flags(self):
```

Comment:
This function exists because I went with the CLI implementation of this. Maybe that was a bad idea.

Reply:
What's the other option? We'd still need to get these args from somewhere right?

Reply:
Invoke dbt via the python method, and pass it the args as part of `self`... I think?

Reply:
Meh I prefer the CLI. I think less coupling is probably better than more coupling here :)